src/admin/content-block/helpers/parsers.ts

```diff
@@ -0,0 +1,62 @@
+import { get } from 'lodash-es';
+
+import { CONTENT_BLOCK_CONFIG_MAP } from '../content-block.const';
+import { ContentBlockConfig, ContentBlockSchema, ContentBlockType } from '../content-block.types';
+
+// Parse content-block config to valid request body
+export const parseContentBlockConfig = (
+	contentBlockConfig: ContentBlockConfig,
+	contentId?: number
+) => {
+	const componentState = contentBlockConfig.components.state;
+	const { blockType, position, ...blockState } = contentBlockConfig.block.state;
+
+	return {
+		position,
+		variables: { componentState, blockState },
+		...(contentId ? { content_id: contentId } : null),
+		content_block_type: blockType,
+	};
+};
+
+export const parseContentBlockConfigs = (
+	contentId: number,
+	contentBlockConfigs: ContentBlockConfig[]
+): Partial<ContentBlockSchema>[] => {
+	const contentBlocks = contentBlockConfigs.map(contentBlockConfig =>
+		parseContentBlockConfig(contentBlockConfig, contentId)
+	);
+
+	return contentBlocks;
```

Comment:
Local variable 'contentBlocks' is redundant

Reply:
Fixed